scopes/component/component/component.ts

```diff
@@ -106,8 +106,12 @@ export class Component {
     }
   }
 
-  async getLogs(filter?: { type?: string; offset?: number; limit?: number }) {
-    const allLogs = await this.factory.getLogs(this.id);
+  async getLogs(filter?: { type?: string; offset?: number; limit?: number; head?: string }) {
+    const id = !filter?.head
```

Comment:
move the condition to the version field

Reply:
I am not sure what you mean by moving it to the version field? 

We only want logs from the head version
Rest of the component data we want according to the version specified in the BitId (which could be different than the head filter here)